### ISSUE TITLE
Error handling for EnvironmentError broken with Scons 3.0.x

### ISSUE THREAD

**Issue by kayhayen** [SCons/scons#3349]
* Version of SCons
3.0.4
* Version of Python
2.7 through 3.7

* How to reproduce your issue?  Please include a small self contained reproducer. Likely a SConstruct should do for most issues.

Again, no SCEE, but the mistake is pretty obvious, in the source code, this is from as Nuitka commit, due to the fork, it's not very clean, but `Errors.py` is wrong here:
```

-    elif isinstance(status, (EnvironmentError, OSError, IOError)):
+    elif isinstance(status, (__builtins__["EnvironmentError"], OSError, IOError)):
```

The `EnvironmentError` defined in that module is not the one that is supposed to be handled there, which is why in error cases, instead of reporting the real issue, an `AttributeError` is raised.

It may be the case that some bug hiding has taken place after 3.0.1 where you started to use getattr, but these appear pretty pointless. Something looks wrong there though, as EnvironmentError as defined, just won't ever fit into these.

**Comment by mwichmann** [comment]
Ugh... it's pretty dodgy for SCons to also use the name `EnvironmentError`, that was bound to lead to confusion.

**Comment by bdbaddog** [comment]
What is "SCEE"?

**Comment by kayhayen** [comment]
My typo version of [Short, Self Contained, Correct, Example](http://sscce.org/) @bdbaddog 

What I am saying is, that I don't have a test case, which I consider bad. But in this instance the flaw in the code is pretty obvious.

**Comment by bdbaddog** [comment]
Fixed in PR #3354
